go/vt/vtgate/scatter_conn.go

```diff
@@ -238,22 +230,17 @@ func (stc *ScatterConn) ExecuteMultiShard(
 	// mu protects qr
 	var mu sync.Mutex
 	qr := new(sqltypes.Result)
-	shards := make([]string, 0, len(shardQueries))
-	for shard := range shardQueries {
-		shards = append(shards, shard)
-	}
 
-	canCommit := len(shards) == 1 && canAutocommit && session.AutocommitApproval()
+	canCommit := len(rss) == 1 && canAutocommit && session.AutocommitApproval()
 
-	err := stc.multiGoTransaction(
+	err := stc.multiGoTransaction2(
```

Comment:
Do we still need the old multigoTransaction?

Reply:
It's still called by vtgate.Execute, which is still called in a couple places. I'll clean that up next, along with the remaining srvtopo methods that are not used any more. This is a huge PR already, so I'll just merge this now, and work on clean up as a separate one.